internal/terraform/node_resource_import.go

```diff
@@ -228,7 +231,7 @@ func (n *graphNodeImportStateSub) Execute(ctx EvalContext, op walkOperation) (di
 			ResolvedProvider: n.ResolvedProvider,
 		},
 	}
-	state, deferred, refreshDiags := riNode.refresh(ctx, states.NotDeposed, state)
+	state, deferred, refreshDiags := riNode.refresh(ctx, states.NotDeposed, state, false)
```

Comment:
nit: just below this there's a check for `deferred != nil`. You could change the error message there to indicate a bug in the provider - we said don't do a deferral and they responded with one, please file a bug with the relevant provider. Or, even just remove the check and trust the providers will behave sensibly.

If you choose to say the whole thing about there being a bug in the provider, we could do the same check above after line 98-100.